sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallDialogAsync.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation;
+
+import com.azure.communication.callautomation.implementation.CallDialogsImpl;
+import com.azure.communication.callautomation.implementation.accesshelpers.DialogStateResponseConstructorProxy;
+import com.azure.communication.callautomation.implementation.models.DialogOptions;
+import com.azure.communication.callautomation.implementation.models.StartDialogRequestInternal;
+import com.azure.communication.callautomation.models.DialogStateResult;
+import com.azure.communication.callautomation.models.StartDialogOptions;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * CallDialogAsync.
+ */
+public final class CallDialogAsync {
+
+    private final CallDialogsImpl dialogsInternal;
+    private final String callConnectionId;
+    private final ClientLogger logger;
+
+    /**
+     * Constructor.
+     *
+     * @param callConnectionId callConnectionId the id of the call connection
+     * @param callDialogInternal Internal implementation of CallDialog
+     */
+    public CallDialogAsync(String callConnectionId, CallDialogsImpl callDialogInternal) {
```

Comment:
don't need public constructor as this is getting retrieved from getCallDialog method from callconnection